lib/web/mage/validation.js

```diff
@@ -1032,7 +1032,7 @@ define([
         ],
         'validate-date': [
             function (value, params, additionalParams) {
-                var test = moment(value, additionalParams.dateFormat);
+                var test = moment(value, additionalParams.dateFormat.toUpperCase());
```

Comment:
Looks like "too much magic" - if I provide `dateFormat` through the argument, I expect to have the date value validated with provided date format. 

Isn't that better to find where the format is injected and change it there instead of doing magic to date format inside validator?

Reply:
Hey @lbajsarowicz 

The Date format for DOB is calculated from this function lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php::getDateFormat and it returns "M/d/Y"

It works well in all places but when it is validated via moment js in below code

var test = moment(value, additionalParams.dateFormat);

moment js do not understand it. It expects it like "M/D/Y" instead. So, I have changed it only for moment js. I hope this makes sense.